lib/kino/video.ex

```diff
@@ -0,0 +1,58 @@
+defmodule Kino.Video do
+  @moduledoc """
+  A kino for rendering a binary video.
+
+  ## Examples
+
+      content = File.read!("/path/to/video.mp4")
+      Kino.Video.new(content, :mp4)
+  """
+
+  use Kino.JS
+
+  @type t :: Kino.JS.t()
+
+  @type mime_type :: binary()
+  @type common_video_type :: :mp4 | :ogg | :avi | :mwv | :mov
+  @doc """
+  Creates a new kino displaying the given binary video.
+
+  The given type be either `:mp4`, `:ogg`, `:avi`, `:wmv`, `:mov`
+  or a string with video MIME type.
+
+  """
+  @spec new(binary(), common_video_type() | mime_type()) :: t()
+  def new(content, type) when is_binary(content) do
+    Kino.JS.new(__MODULE__, data_url(content, type))
```

Comment:
Encoding as data URL inflates the size. Instead we can send the raw binary:

```suggestion
    Kino.JS.new(__MODULE__, {:binary, content, %{type: type}})
```

On the client we receive it as `function init(ctx, [content, { type }]) {...}`.

To convert the raw binary to base64 on the client, you can use this:

```js
function bufferToBase64(buffer) {
  let binaryString = "";
  const bytes = new Uint8Array(buffer);
  const length = bytes.byteLength;

  for (let i = 0; i < length; i++) {
    binaryString += String.fromCharCode(bytes[i]);
  }

  return btoa(binaryString);
}
```

Reply:
```elixir 
def new(content, type) when is_binary(content) do
  Kino.JS.new(__MODULE__, {:binary, content, %{type: mime_type!(type)}})
end
```
don't work for me geting:
```
Failed to serialize initial widget data, invalid byte 0xE6 in <<82, 73,
```

Reply:
Ah sorry, we only support this for `Kino.JS.Live` indeed. That said, I would actually make it `Kino.JS.Live` that stores content in the state and sends that payload on `handle_connect`. It means we use an additional process, but the advantage is that the binary is sent over a separate channel than LV operates on. For audio/video outputs it's more likely that the content is larger, so using a separate channel is better.